xbmc/music/MusicUtils.cpp

```diff
@@ -177,42 +181,84 @@ namespace MUSIC_UTILS
     CJobManager::GetInstance().AddJob(job, NULL);
   }
 
-  bool FillArtTypesList(CFileItem& musicitem, CFileItemList& artlist)
+  // Add art types required in Kodi
+  void AddHardCodedArtTypes(std::vector<std::string>& artTypes, const CMusicInfoTag& tag)
   {
-    CMusicInfoTag &tag = *musicitem.GetMusicInfoTag();
-    if (tag.GetDatabaseId() < 1 || tag.GetType().empty())
-      return false;
-    if (tag.GetType() != MediaTypeArtist && tag.GetType() != MediaTypeAlbum && tag.GetType() != MediaTypeSong)
-      return false;
-
-    artlist.Clear();
-    // Songs, albums and artists all  have thumbs by default
-    std::vector<std::string> artTypes = { "thumb" };
+    artTypes.emplace_back("thumb");
     if (tag.GetType() == MediaTypeArtist)
     {
       artTypes.emplace_back("fanart");
     }
+  }
 
-    CMusicDatabase db;
-    db.Open();
+  // Add art types configured by the user
+  void AddExtendedArtTypes(std::vector<std::string>& artTypes, const CMusicInfoTag& tag)
+  {
+    for (const auto& artType : CMusicInfoScanner::GetArtTypesToScan(tag.GetType()))
```

Comment:
GetArtTypesToScan is not in CMusicInfoScanner any more!
I suggest you correct and then squash the commits